nats.go

```diff
@@ -50,8 +50,7 @@ const (
 	Version                   = "1.24.0"
 	DefaultURL                = "nats://127.0.0.1:4222"
 	DefaultPort               = 4222
-	DefaultMaxReconnect       = 60
-	DefaultReconnectWait      = 2 * time.Second
+	DefaultMaxReconnect       = -1
```

Comment:
makes sense to change into a span of time but not sure about changing this var as the default since has been like that from the start... maybe as an option to test out new behavior like `MaxTotalReconnectTime(2*time.Minute)`? We also need to be able to set `max reconnect` to -1 so that it means that we want the client to reconnect forever and never give up reconnecting

Reply:
@piotrpio's change is right in line with this.